app/models/notification.rb

```diff
@@ -79,7 +83,22 @@ def unarchive_if_updated
   def update_from_api_response(api_response, unarchive: false)
     attrs = Notification.attributes_from_api_response(api_response)
     self.attributes = attrs
+    set_author(api_response)
     unarchive_if_updated if unarchive
     save(touch: false) if changed?
   end
+
+  private
+
+  def set_author(api_response)
+    /\/((?:issues|pulls)\/(?<issue_number>\d+))|((?:commits)\/(?<commit_sha>[0-9a-f]{5,40}))\z/ =~ api_response[:subject][:url]
```

Comment:
That's a helluva regex, but it makes sense 😄

Could we perhaps make that a constant and put it at the top of the model?

```ruby
NOTIFICATION_SUBJECT_TYPE = %r{\/((?:issues|pulls)\/(?<issue_number>\d+))|((?:commits)\/(?<commit_sha>[0-9a-f]{5,40}))\z}

# ....

private

def set_author(api_response)
  NOTIFICATION_SUBJECT_TYPE =~ api_response[:subject][:url]
  # ....
end
```